lib/plug/conn.ex

```diff
@@ -460,12 +460,15 @@ defmodule Plug.Conn do
   otherwise `{:error, reason}`.
   """
   @spec chunk(t, body) :: {:ok, t} | {:error, term} | no_return
-  def chunk(%Conn{state: :chunked} = conn, ""), do: {:ok, conn}
   def chunk(%Conn{adapter: {adapter, payload}, state: :chunked} = conn, chunk) do
-    case adapter.chunk(payload, chunk) do
-      :ok                  -> {:ok, conn}
-      {:ok, body, payload} -> {:ok, %{conn | resp_body: body, adapter: {adapter, payload}}}
-      {:error, _} = error  -> error
+    case IO.iodata_length(chunk) do
```

Comment:
@myronmarston I had a couple string sizes, but included @lexmag's large string to the list and got:

```
## BasicBench
benchmark name       iterations   average time 
empty_iodata?          10000000   0.76 µs/op
IO.iodata_length           1000   1284.00 µs/op
IO.iodata_to_binary         500   4616.14 µs/op

```